vm-setup/roles/v1a2_integration_test/tasks/deprovision.yml

```diff
@@ -0,0 +1,39 @@
+---
+  - name: Deprovision machines
+    k8s:
+      state: absent
+      api_version: "{{ API_VERSION }}"
+      kind: Machine
+      namespace: "{{ NAMESPACE }}"
+      name: "{{ item }}"
+    loop:
+       - "{{ CLUSTER_NAME }}-controlplane-0"
+       - "{{ CLUSTER_NAME }}-md-0"
+    register: deprovision_machines
+
+  - name: Deprovision cluster
+    k8s:
+      state: absent
+      api_version: "{{ API_VERSION }}"
+      kind: Cluster
+      namespace: "{{ NAMESPACE }}"
+      name: "{{ CLUSTER_NAME }}"
+    register: deprovision_cluster
+
```

Comment:
you should also remove manually the kubeadmconfigTemplate and the baremetalmachinetemplate that are not automatically deleted